fuel-txpool/src/service.rs

```diff
@@ -40,36 +48,75 @@ impl Default for ServiceBuilder {
 
 impl ServiceBuilder {
     pub fn new() -> Self {
-        let (sender, receiver) = mpsc::channel(100);
-        let (broadcast, _receiver) = broadcast::channel(100);
         Self {
-            sender: txpool::Sender::new(sender),
-            receiver,
-            db: None,
-            broadcast,
             config: Default::default(),
-            import_block_events: None,
+            db: None,
+            txpool_sender: None,
+            txpool_receiver: None,
+            tx_status_sender: None,
+            import_block_receiver: None,
+            incoming_tx_receiver: None,
+            #[cfg(feature = "p2p")]
+            network_sender: None,
         }
     }
 
     pub fn sender(&self) -> &txpool::Sender {
-        &self.sender
+        self.txpool_sender.as_ref().unwrap()
     }
 
     pub fn subscribe(&self) -> broadcast::Receiver<TxStatusBroadcast> {
-        self.broadcast.subscribe()
+        self.tx_status_sender.as_ref().unwrap().subscribe()
     }
 
     pub fn db(&mut self, db: Box<dyn TxPoolDb>) -> &mut Self {
         self.db = Some(db);
         self
     }
 
+    pub fn txpool_sender(&mut self, txpool_sender: txpool::Sender) -> &mut Self {
+        self.txpool_sender = Some(txpool_sender);
+        self
+    }
+
+    pub fn txpool_receiver(
+        &mut self,
+        txpool_receiver: mpsc::Receiver<TxPoolMpsc>,
+    ) -> &mut Self {
+        self.txpool_receiver = Some(txpool_receiver);
+        self
+    }
+
+    pub fn tx_status_sender(
+        &mut self,
+        tx_status_sender: broadcast::Sender<TxStatusBroadcast>,
+    ) -> &mut Self {
+        self.tx_status_sender = Some(tx_status_sender);
+        self
+    }
+
+    pub fn incoming_tx_receiver(
+        &mut self,
+        incoming_tx_receiver: broadcast::Receiver<TransactionBroadcast>,
+    ) -> &mut Self {
+        self.incoming_tx_receiver = Some(incoming_tx_receiver);
+        self
+    }
+
+    #[cfg(feature = "p2p")]
```

Comment:
could we remove this feature flag from txpool + block producer and inject a dummy noop impl of this sender when the central p2p feature is disabled from fuel-core? Otherwise these feature flags will start proliferating all over the place.

Reply:
Yeah, I can try that out. The feature flag switches really do spread like rabbits once they're in there.

Reply:
Should be fixed now